### PR TITLE
[#1460] Handle shutdown scenario properly

### PR DESCRIPTION
**Pull request by szarnekow**
[#1460] Handle shutdown scenario properly

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/WorkspaceAccess.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@SuppressWarnings("restriction")
+public interface WorkspaceAccess {
+	
+	enum Result {
+		YES,
+		NO,
+		SHUTDOWN
+	}
+
+	static Result isWorkspaceLockedByCurrentThread(IWorkspace workspace) {
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/WorkspaceAccess.java]: am not sure if i like the static here (for patchability)

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/WorkspaceAccess.java]: I'll make this injectable.

File: org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/SharedModule.java [hunk 2]
```diff
@@ -203,6 +204,7 @@ protected void configure() {
 				expose(ReferenceSearchViewPage.class);
 			}
 		});
+		bind(WorkspaceLockAccess.class);
```
Comment (cdietrich) [inline org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/SharedModule.java]: i wonder if we have any scheme for ordering the stuff here

Comment (szarnekow) [reply org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/SharedModule.java]: I did not find an obvious spot, so I put it at the end.

**Pull request merged**